pkg/utils/utils.go

```diff
@@ -632,7 +660,86 @@ func GetClusterFlavor(config *rest.Config) (constants.ClusterFlavor, error) {
 		return constants.Unknown, err
 	}
 
-	return GetCSIClusterType(clientset)
+	veleroNs, exist := os.LookupEnv("VELERO_NAMESPACE")
+	if !exist {
+		fmt.Printf("The VELERO_NAMESPACE environment variable is empty, assuming velero as namespace\n")
+		veleroNs = "velero"
+	}
+
+	return retrieveClusterFlavor(clientset, veleroNs)
+}
+
+// Internal method to enable unit testing.
```

Comment:
Add an extra func `retrieveClusterFlavor` is unnecessary. Unit test can be run directly against RetrieveClusterFlavor. 

Please double check the test case at https://github.com/vmware-tanzu/velero-plugin-for-vsphere/blob/8ce725659a49cce917af17e819933a9f5d2c801d/pkg/utils/utils_unit_test.go#L172 for an example.

Reply:
This is nice, thanks for pointing it out.

Reply:
Just FYI, i've fixed this for `RetrieveVcConfigSecret`, however, `retrieveClusterFlavor` is a little different. Here, the method `RetrieveClusterFlavor` is called with veleroNs passed on, the velero ns is passed because for some reason during the backup-driver & data-manager startup the environment variable `VELERO_NAMESPACE` is not found, I suspect this has to do something with how velero starts adds the plugin. `RetrieveClusterFlavor` is exclusively called  from `install.go` of backup-driver and data-manager, while the rest of the code in plugin determines the cluster flavor using `GetClusterFlavor`. Both these methods eventually call `retrieveClusterFlavor` for which I've written unit tests.